src/OpenTelemetry.Exporter.OpenTelemetryProtocol/OtlpExporterOptions.cs

```diff
@@ -101,23 +115,29 @@ public Uri Endpoint
         set
         {
             this.endpoint = value;
-            this.AppendSignalPathToEndpoint = false;
+            this.AppendSignalPathToEndpoint = value == null;
```

Comment:
Could we throw an exception instead? Why allow the user to even set the Endpoint to `null` in the first place?

Reply:
We could throw. I thought this would be the most flexible thing to do. Say `Endpoint` got set from some other library or some config file but in my code I want to force the defaults. The code as-is, I could always do:

```csharp
services.PostConfigure<OtlpExporterOptions>(o =>
{
   o.Endpoint = null!; // Use default behavior regardless of what was done prior
});
```

Reply:
The `o.Endpoint = null!;` to reset defaults seems too hacky. For such cases it shouldn't be hard for someone to just provide the default endpoint as is:

```csharp
services.PostConfigure<OtlpExporterOptions>(o =>
{
   o.Endpoint = new Uri("http://localhost:4318/v1/traces"); // Use default behavior regardless of what was done prior
});
```

Reply:
> The o.Endpoint = null!; to reset to default seems too hacky

+1

Reply:
Updated: https://github.com/open-telemetry/opentelemetry-dotnet/pull/5434/commits/80871d3b4135d2413ee07fe7474274585a8dee15